airbyte-integrations/connectors/source-shopify/source_shopify/utils.py

```diff
@@ -63,9 +79,33 @@ class ShopifyRateLimiter:
     on_high_load: float = 5.0
 
     @staticmethod
-    def get_wait_time(*args, threshold: float = 0.9, rate_limit_header: str = "X-Shopify-Shop-Api-Call-Limit"):
+    def _convert_load_to_time(load: Optional[int], threshold: float) -> float:
         """
-        To avoid reaching Shopify API Rate Limits, use the "X-Shopify-Shop-Api-Call-Limit" header value,
+            Define wait_time based on load conditions.
+
+            :: load - represents how close we are to being throttled
+                    - 0.5 is half way through our allowance
+                    - 1 indicates that all of the allowance is used and the api will start rejecting calls
+            :: threshold - is the % cutoff for the rate_limits/load
+            :: wait_time - time to wait between each request in seconds
+
+        """
+        mid_load = threshold / 2  # average load based on threshold
+        if not load:
+            # when there is no rate_limits from header, use the `sleep_on_unknown_load`
+            wait_time = ShopifyRateLimiter.on_unknown_load
+        elif load >= threshold:
+            wait_time = ShopifyRateLimiter.on_high_load
+        elif load >= mid_load:
+            wait_time = ShopifyRateLimiter.on_mid_load
+        elif load < mid_load:
+            wait_time = ShopifyRateLimiter.on_low_load
+        return wait_time
+
+    @staticmethod
+    def get_rest_api_wait_time(*args, threshold: float = 0.9, rate_limit_header: str = "X-Shopify-Shop-Api-Call-Limit"):
```

Comment:
line long

Reply:
This also was unaffected by `black -l 120 . --extend-exclude virtualenv`.